huaweicloud/services/cpts/resource_huaweicloud_cpts_task.go

```diff
@@ -0,0 +1,293 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	v1 "github.com/huaweicloud/huaweicloud-sdk-go-v3/services/cpts/v1"
+	"github.com/huaweicloud/huaweicloud-sdk-go-v3/services/cpts/v1/model"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+)
+
+const (
+	clusterTypeShare   = "shared-cluster-internet"
+	clusterTypePrivate = "private-cluster"
+
+	runStatusRunning   = 0
+	runStatusFinished  = 2
+	runStatusToRunning = 9
+
+	operationEnable = "enable"
+	operationStop   = "stop"
+)
+
+func ResourceTask() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceTaskCreate,
+		UpdateContext: resourceTaskUpdate,
+		DeleteContext: resourceTaskDelete,
+		ReadContext:   resourceTaskRead,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Update: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 42),
+			},
+
+			"project_id": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"benchmark_concurrency": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  100,
+			},
+
+			"cluster_id": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+
+			"operation": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{operationEnable, operationStop}, false),
+			},
+
+			"status": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceTaskCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	c := meta.(*config.Config)
+	region := c.GetRegion(d)
+	client, err := c.HcCptsV3Client(region)
+	if err != nil {
+		return diag.Errorf("error creating CPTS v1 client: %s", err)
+	}
+
+	projectId := int32(d.Get("project_id").(int))
+	benchConcurrent := int32(d.Get("benchmark_concurrency").(int))
+	createOpts := &model.CreateTaskRequest{
+		Body: &model.CreateTaskRequestBody{
+			Name:            d.Get("name").(string),
+			ProjectId:       projectId,
+			BenchConcurrent: &benchConcurrent,
+		},
+	}
+
+	response, err := client.CreateTask(createOpts)
+	if err != nil {
+		return diag.Errorf("error creating CPTS task: %s", err)
+	}
+
+	if response.TaskId == nil {
+		return diag.Errorf("error creating CPTS task: id not found in api response")
+	}
+
+	d.SetId(strconv.Itoa(int(*response.TaskId)))
+	return resourceTaskRead(ctx, d, meta)
+}
+
+func resourceTaskRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
replace ctx with _ as it's not used in this func

Reply:
fix